string source
float32 current
float32 target
